tpu-client-next/src/connection_workers_scheduler.rs

```diff
@@ -338,21 +337,23 @@ impl WorkersBroadcaster for NonblockingBroadcaster {
     }
 }
 
-/// Splits `leaders` into two slices based on the `fanout` configuration:
-/// * the first slice contains the leaders to which transactions will be sent,
-/// * the second vector contains the leaders, used to warm up connections. This
-///   slice includes the first set.
-fn split_leaders<'leaders>(
-    leaders: &'leaders [SocketAddr],
-    fanout: &Fanout,
-) -> (&'leaders [SocketAddr], &'leaders [SocketAddr]) {
-    let Fanout { send, connect } = fanout;
-    assert!(send <= connect);
-    let send_count = (*send).min(leaders.len());
-    let connect_count = (*connect).min(leaders.len());
-
-    let send_slice = &leaders[..send_count];
-    let connect_slice = &leaders[..connect_count];
-
-    (send_slice, connect_slice)
+/// Extracts a list of unique leader addresses to which transactions will be sent.
+///
+/// This function selects up to `send_fanout` addresses from the `leaders` list, ensuring that
+/// only unique addresses are included while maintaining their original order.
+fn extract_send_leaders(leaders: &[SocketAddr], send_fanout: usize) -> Vec<SocketAddr> {
+    let send_count = send_fanout.min(leaders.len());
+    remove_duplicates(&leaders[..send_count])
```

Comment:
should we not inline remove_duplicates?

Reply:
Have nothing against it, do you think it is enough to use `#[inline]` (or always)? Yet the impact should be small, because this function call maybe takes 2ns while the execution of the function ~40ns.

Reply:
this will get auto-inlined 100% since there is only one caller. it is more a question of style=)